src/items.js

```diff
@@ -1,291 +1,153 @@
+        { action: moreActionsButton },
+      ];
+    }
 
-    return actionDiv;
+    function itemFooter() {
+      const tokenArea = components.newTextArea({
+        id: `token-${item.id}`,
+        classNames: "tokenTextArea w-100 display-none",
+        style: "overflowX: hidden; overflowY: scroll",
+        text: item.value,
+        disabled: true,
+      });
+
+      const applyButton = components.newButton({
+        classNames: "btn btn-outline-success",
+        onClick: () => applyKey(item.id),
+        label: "Apply",
+        icon: {
+          classNames: "fa-solid fa-floppy-disk fa-lg me-1",
+        },
+      });
+      return {
+        body: tokenArea,
+        actions: [applyButton],
+      };
+    }
   }
 
-  function newDropdown(itemId, value, actions) {
-    const dropdown = document.createElement("div");
-    dropdown.classList.add("dropdown");
-    dropdown.classList.add("dropstart");
-    dropdown.append(newDropDownButton());
-
-    const options = document.createElement("ul");
-    options.classList.add("dropdown-menu");
-
-    options.appendChild(
-      newDropdownOption(
-        itemId,
-        "View value",
-        !value,
-        {
-          class: "fa-solid fa-eye fa-lg",
-          style: "color: #495057; margin-right: 10px;",
+  function moreActions({ id, value, actions }) {
+    const options = [
+      {
+        id: `viewBtn-${id}`,
+        itemId: id,
+        icon: {
+          classNames: "fa-solid fa-eye fa-lg me-2",
+          style: "color: var(--dark-gray);",
         },
-        actions.viewFnc,
-        `viewBtn-${itemId}`
-      )
-    );
-    options.appendChild(
-      newDropdownOption(
-        itemId,
-        "Set value",
-        false,
-        {
-          class: "fa-solid fa-wand-magic-sparkles fa-lg",
-          style: "color: #495057; margin-right: 10px;",
+        label: "View value",
+        disabled: !value,
+        onClick: actions.viewFnc,
+      },
+      {
+        id: `setBtn-${id}`,
+        itemId: id,
+        icon: {
+          classNames: "fa-solid fa-wand-magic-sparkles fa-lg me-2",
+          style: "color: var(--dark-gray);",
         },
-        function () {
-          const textArea = document.getElementById(`token-${itemId}`);
-          textArea.classList.remove("display-none");
-          textArea.disabled = false;
-          const textAreaFooter = document.getElementById(
-            `textAreaFooter-${itemId}`
-          );
-          textAreaFooter.classList.remove("display-none");
-        }
-      )
-    );
-    options.appendChild(newLiItemSeparator());
-    options.appendChild(
-      newDropdownOption(
-        itemId,
-        "Delete key",
-        false,
-        {
-          class: "fa-solid fa-trash-can fa-lg",
-          style: "color: #495057; margin-right: 10px;",
+        label: "Set value",
+        disabled: false,
+        onClick: () => showApplyFooter(id),
+        separator: true,
+      },
+      {
```

Comment:
```suggestion
      },
      {
        separator: true,
      },
      {
```

Even though we discussed offline about having the separator as part of one of the options, wouldn't make sense to have a dedicate "option" as separator ?

Reply:
As discussed offline, I don't agree with having a separator as a dropdown option because it isn't a dropdown option. If anything, a "real" option should tell the menu if it should be separated from the other options, that's why we created the separator prop. Anyway, I moved this separator validation to be applied on top of the option and not bottom as before.